### PR TITLE
feat: [CDS-83039] Suggest hint to remove templates from the path when specifying helm chart

### PR DESCRIPTION
**Pull request by alebuh**
## Describe your changes
When the helm template fails, add additional hint 
Screenshot 2023-11-28 at 15 28 08
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56070)

### CODE REVIEW

**Comment by alebuh** [comment]
trigger uts

**Review by abhinavsingh1196 — APPROVED** [review]

File: 970-api-services-beans/src/main/java/io/harness/k8s/exception/KubernetesExceptionHints.java [hunk 1]
```diff
@@ -102,7 +102,7 @@ public class KubernetesExceptionHints {
   public final String MANIFEST_RENDER_ERROR_GO_TEMPLATE =
       "Please check if the given manifests are valid and can be rendered by Go template. To know more about Go templating, refer: \nhttps://community.harness.io/t/harness-local-go-templating/460 \nhttps://docs.harness.io/article/mwy6zgz8gu-use-go-templating-in-kubernetes-manifests#notes";
   public final String MANIFEST_RENDER_ERROR_HELM =
-      "Please check if the given manifests are valid and can be rendered by the `helm template` command.";
+      "Please check if the given manifests are valid and can be rendered by the `helm template` command. \n- The templates folder should not be a part of the specified path for the helm chart.\nFor more info check the chart file structure: https://helm.sh/docs/topics/charts/#the-chart-file-structure";
```
Comment (abosii-harness) [inline 970-api-services-beans/src/main/java/io/harness/k8s/exception/KubernetesExceptionHints.java:105]: The hint should be based on if customer provided templates folder in path or not

**Review by abosii-harness — CHANGES_REQUESTED** [review]

File: 970-api-services-beans/src/main/java/io/harness/k8s/exception/KubernetesExceptionHints.java (hunk 1 above)
Comment (alebuh) [reply 970-api-services-beans/src/main/java/io/harness/k8s/exception/KubernetesExceptionHints.java:105]: @abosii-harness At the moment when we are throwing this exception, we don't have information about the provided path as we are using a working directory where we already copied all files from the path. Currently, we can't know if the user provided templates in the path. We can expand the method signature of: io.harness.delegate.task.k8s.K8sTaskHelperBase#renderTemplateForHelm(java.lang.String, java.lang.String, java.util.List<java.lang.String>, java.lang.String, java.lang.String, io.harness.logging.LogCallback, io.harness.k8s.model.HelmVersion, long, io.harness.delegate.task.helm.HelmCommandFlag, java.lang.String, java.lang.String)
by adding one more param that would be provided path. But then we would need to add logic for each StoreDelegateConfig for extracting path value.
Let me know if it's worth doing this just to expand the hint message, or if there is any other way to achieve this that I'm missing?

**Review by abosii-harness — APPROVED** [review]

**Comment by alebuh** [comment]
trigger review

**Pull request merged**